compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -393,11 +405,35 @@ trait QuotesAndSplices {
       case Some(argPt: ValueType) => argPt // excludes TypeBounds
       case _ => defn.AnyType
     }
-    val quoted0 = desugar.quotedPattern(quoted, untpd.TypedSplice(TypeTree(quotedPt)))
-    val quoteCtx = quoteContext.addMode(Mode.QuotedPattern).retractMode(Mode.Pattern)
-    val quoted1 =
-      if quoted.isType then typedType(quoted0, WildcardType)(using quoteCtx)
-      else typedExpr(quoted0, WildcardType)(using quoteCtx)
+    val (untpdTypeVariables, quoted0) = desugar.quotedPatternTypeVariables(desugar.quotedPattern(quoted, untpd.TypedSplice(TypeTree(quotedPt))))
+
+    for untpdTypeVariable <- untpdTypeVariables do untpdTypeVariable.rhs match
+      case _: TypeBoundsTree => // ok
+      case LambdaTypeTree(_, body: TypeBoundsTree) => // ok
+      case _ => report.error("Quote type variable definition cannot be an alias", untpdTypeVariable.srcPos)
```

Comment:
If we declare a type variable with the same as an alias, then we know the type of that variable. It would be useless to solve the type of that type variable at runtime when pattern matching. This could also be seen as a missed optimization.